File: apps/mobile/lib/data/mock/demo_notifications.dart
import '../../core/constants/statuses.dart';
import '../models/app_notification.dart';

DateTime _ago(int minutes) => DateTime.now().subtract(Duration(minutes: minutes));

/// Yolcu için örnek bildirimler (senaryo).
final List<AppNotification> demoNotifications = [
  AppNotification(
    id: 'ntf-1',
    title: 'Servis başladı',
    message: 'Avrupa Yakası Sabah Servisi yola çıktı.',
    type: NotificationType.tripStarted,
    readAt: null,
    createdAt: _ago(8),
  ),
  AppNotification(
    id: 'ntf-2',
    title: 'Servis gecikiyor',
    message: 'Trafik nedeniyle yaklaşık 3 dakika gecikme bekleniyor.',
    type: NotificationType.delayed,
    readAt: null,
    createdAt: _ago(5),
  ),
  AppNotification(
    id: 'ntf-3',
    title: 'Servis 5 durak uzakta',
    message: 'Aracınız durağınıza 5 durak uzaklıkta.',
    type: NotificationType.fiveStopsAway,
    readAt: _ago(30),
    createdAt: _ago(30),
  ),
];

File: apps/mobile-legacy-rn/src/mocks/demoNotifications.ts
import { NOTIFICATION_TYPE, type AppNotification } from '@servis/shared';
import { DEMO_TENANT_ID, DEMO_PASSENGER_ID } from './demoUsers';

function iso(minutesAgo: number): string {
  return new Date(Date.now() - minutesAgo * 60000).toISOString();
}

/** Yolcu için örnek bildirimler (senaryo). */
export const demoNotifications: AppNotification[] = [
  {
    id: 'ntf-1',
    tenant_id: DEMO_TENANT_ID,
    user_id: DEMO_PASSENGER_ID,
    title: 'Servis başladı',
    message: 'Avrupa Yakası Sabah Servisi yola çıktı.',
    type: NOTIFICATION_TYPE.TRIP_STARTED,
    data: null,
    read_at: null,
    created_at: iso(8),
  },
  {
    id: 'ntf-2',
    tenant_id: DEMO_TENANT_ID,
    user_id: DEMO_PASSENGER_ID,
    title: 'Servis gecikiyor',
    message: 'Trafik nedeniyle yaklaşık 3 dakika gecikme bekleniyor.',
    type: NOTIFICATION_TYPE.DELAYED,
    data: { delay_minutes: 3 },
    read_at: null,
    created_at: iso(5),
  },
  {
    id: 'ntf-3',
    tenant_id: DEMO_TENANT_ID,
    user_id: DEMO_PASSENGER_ID,
    title: 'Servis 5 durak uzakta',
    message: 'Aracınız durağınıza 5 durak uzaklıkta.',
    type: NOTIFICATION_TYPE.FIVE_STOPS_AWAY,
    data: null,
    read_at: iso(30),
    created_at: iso(30),
  },
];
